### ISSUE TITLE
Server Closes After Setting Up Worker Threads

### ISSUE THREAD

**Issue by BreadDisease** [SmartlyDressedGames/U3-Docs#115]
Hello,

My server closes right after saying "Loading miscellaneous..." and 
"Setting up 6 worker threads for Enlighten.
  Thread -> id: 2214 -> priority: 1 
  Thread -> id: 12f0 -> priority: 1 
  Thread -> id: fbc -> priority: 1 
  Thread -> id: 1a64 -> priority: 1 
  Thread -> id: 1624 -> priority: 1 
  Thread -> id: 8d0 -> priority: 1"

System is running R5 3600, 16 GB, GTX 760, W10 21H1. I can launch the game fine, but trying to start the server either using the SteamCMD installation or the Unturned Dedicated Server results in the same crash happening. 

Logs of server startup are attached. Any help is much appreciated. Thanks

[crash.txt](https://github.com/SmartlyDressedGames/U3-Docs/files/7766475/crash.txt)

**Comment by SDGNelson** [comment]
Is there a "crash.dmp" adjacent to the log file? If so, could you send it to me please?

**Comment by BreadDisease** [comment]
Where would that be? There's no crash.dmp inside the U3DS folder. The log file I sent I obtained by using "ExampleServer.bat > crash.txt" in command prompt.

**Comment by SDGNelson** [comment]
Ah, sorry, I thought you found crash.txt alongside the other crash files. The crashes folder should be here:
"%TEMP%\Smartly Dressed Games\Unturned"

**Comment by BreadDisease** [comment]
There's no such folder Smartly Dressed Games in %TEMP%. I launched the server again to make sure and there's nothing there

**Comment by SDGNelson** [comment]
In that case I wonder if it is either maybe not actually crashing, or if other software like anti-virus is closing the game. If you can attach the Server_ (id) .log file from the U3DS/Logs folder there might be something.

**Comment by BreadDisease** [comment]
I don't have an AV other than Windows Defender. Disabling Realtime Protection for Defender didn't change anything. I've attached log file. Thanks!

[Server_Example.log](https://github.com/SmartlyDressedGames/U3-Docs/files/7771722/Server_Example.log)

**Comment by BreadDisease** [comment]
Looking in the logs, I found

"Quit due to provider exception: GameServer API initialization failed!"

I was reading a bunch of other threads talking about that issue on Linux but I'm on Windows 10. Any ideas?

**Comment by BreadDisease** [comment]
I tried running the Unturned Dedicated Server on my PC (on the same network running W10) instead of my server and that worked. I tried disabling Windows Firewall on my server and it still closes with the same error. Is there a prerequisite that maybe my server doesn't have?

**Comment by SDGNelson** [comment]
Usually that error means the port is already in use, but fairly recently there was a report where it turned out steamcmd was out of date because they were using LGSM. Do either of those sound possible for your setup? e.g. are you using a third-party server management tool?

**Comment by BreadDisease** [comment]
I figured it out! If I changed the port to something else other than the default of 27015 the server would start. Then I used netstat -nao|findstr 27015 to find the PID of the process binding to port 27015. It turns out that an old ARK server was auto-starting, but not showing up in a CMD window like it normally would. After removing the server from auto-starting the server now works with the default port!

Thanks Nelson!
